File: C/boletimNotas.c
/*
# Faça um Programa em Python que 
Leia uma matriz 10 x 3 com as notas 
de 10 alunos em 3 provas. 
Em seguida, escreva o numero de alunos cuja pior nota foi na prova 1, o número de alunos cuja pior nota foi na prova 2, e o numero de alunos cuja pior nota foi na prova 3. Em caso de empate das piores notas de um aluno, o critério de desempate  e arbitrário, mas o aluno deve ser contabilizado apenas uma vez.
*/

#include<stdio.h>
#include<stdlib.h>

void main(){
    int vectorAlunoProva[3][10];
    int numerosDeAluonosPiorNotaProva01 =0;
    int numerosDeAluonosPiorNotaProva02 =0;
    int numerosDeAluonosPiorNotaProva03 =0;
    int piorNota =0;
    int alunos = 1;

    for(int indexProva=0; indexProva<3; indexProva++){
        for(int indexAluno=0; indexAluno<10; indexAluno++){
            printf("Digite a nota do %i aluno da prova %i: ", alunos,indexProva);
            scanf("%d", &vectorAlunoProva[indexProva][indexAluno]);
            alunos++;

        }
    }

    piorNota = vectorAlunoProva[0][0];

    
    for(int indexAluno=0; indexAluno<10; indexAluno++){
           
       piorNota = vectorAlunoProva[0][indexAluno];
       if( piorNota >= vectorAlunoProva[1][indexAluno] ){
            piorNota = vectorAlunoProva[1][indexAluno];
        }
        if(piorNota >= vectorAlunoProva[2][indexAluno]){
            piorNota = vectorAlunoProva[2][indexAluno];
        }
        
        if(piorNota == vectorAlunoProva[0][indexAluno]){
            numerosDeAluonosPiorNotaProva01++;
        }
        if(piorNota == vectorAlunoProva[1][indexAluno]){
            numerosDeAluonosPiorNotaProva02++;
        }
        else{
            numerosDeAluonosPiorNotaProva03++;
        }
    }
    
    printf("\n O numero de alunos com pior nota na 1ª prova: %i",numerosDeAluonosPiorNotaProva01);
    printf("\n O numero de alunos com pior nota na 2ª prova: %i",numerosDeAluonosPiorNotaProva02);
    printf("\n O numero de alunos com pior nota na 3ª prova: %i",numerosDeAluonosPiorNotaProva03);
    

}

File: Java/src/BoletimNotas.java
public class BoletimNotas {

    public static void main(String[] args) {
        int numberAlunos = 10;
        int numberProva = 3;
        int[][] matrizAlunoProva = new int[numberProva][numberAlunos]; 

        int piorNota = 100;
        int numerosDeAluonosPiorNotaProva01 = 0;
        int numerosDeAluonosPiorNotaProva02 = 0;
        int numerosDeAluonosPiorNotaProva03 = 0;

        int alunos = 1;

        InputKeyboard inputKeyboard = new InputKeyboard();

        for (int i = 0; i < matrizAlunoProva.length; i++) {
            for (int j = 0; j < matrizAlunoProva.length; j++) {
                System.out.print("Digite a nota do "+alunos+"aluno na prova "+i+" :");
                int notaProva = inputKeyboard.readInt();
                matrizAlunoProva[i][j] = notaProva;
                alunos++;
            }
            
        }

        for (int j = 0; j < matrizAlunoProva.length; j++) {
            
            piorNota = matrizAlunoProva[0][j];

            if(piorNota >= matrizAlunoProva[1][j]) { 
                piorNota = matrizAlunoProva[1][j];
            }
            if( piorNota >= matrizAlunoProva[2][j]) {
                piorNota = matrizAlunoProva[2][j];
            }

            if(piorNota == matrizAlunoProva[0][j]) {
                numerosDeAluonosPiorNotaProva01++;
            }
            if(piorNota == matrizAlunoProva[1][j]) {
                numerosDeAluonosPiorNotaProva02++;
            }
            else{
                numerosDeAluonosPiorNotaProva03++;
            }
        }

        System.out.println("\n\nO numero de alunos com pior nota na 1ª prova: "+numerosDeAluonosPiorNotaProva01);
        System.out.println("O numero de alunos com pior nota na 2ª prova: "+numerosDeAluonosPiorNotaProva02);
        System.out.println("O numero de alunos com pior nota na 3ª prova: "+numerosDeAluonosPiorNotaProva03);
        
    } 
    
}
